activesupport/test/broadcast_logger_test.rb

```diff
@@ -307,6 +307,84 @@ def info(msg, &block)
       assert_equal true, @logger.error("Hello")
     end
 
+    test "with_level invokes its block one time, even when broadcasting to 2 or more loggers" do
+      invocations = 0
+      log1 = ::Logger.new(nil)
+      log2 = ::Logger.new(nil)
+
+      logger = BroadcastLogger.new(log1, log2)
+
+      invocations = 0
+      logger.with_level(:info) do
+        invocations += 1
+      end
+
+      assert_equal 1, invocations
+    end
+
+    test "with_level invokes its block one time, even when broadcasting to zero loggers" do
+      logger = BroadcastLogger.new()
+
+      invocations = 0
+      logger.with_level(:info) do
+        invocations += 1
+      end
+
+      assert_equal 1, invocations
+    end
+
+    test "with_level returns nil" do
+      logger = BroadcastLogger.new(
+        ::Logger.new(StringIO.new),
+        ::Logger.new(StringIO.new)
+      )
+
+      assert_nil logger.with_level(:debug) { :foo }
+    end
+
+    test "with_level changes the logging level, and restores it afterwards" do
+      buf1 = StringIO.new
+      buf2 = StringIO.new
+
+      log1 = ::Logger.new(buf1)
+      log2 = ::Logger.new(buf2)
+
+      logger = BroadcastLogger.new(log1, log2)
+      logger.formatter = proc do |severity, time, progname, message|
+        "#{severity}:#{message}\n"
+      end
+
+      logger.level = :info
+      logger.debug("AAA") # skipped
+
+      logger.with_level(:debug) do
+        # Inside, debug logging should be captured
+        assert_equal(0, logger.level)
+
+        logger.debug("BBB") # captured
+        logger.info("CCC") # captured
+
+        logger.with_level(:error) do
+          assert_equal(3, logger.level)
```

Comment:
On `main`, this test fails here, the reason is a little unintuitive:

When this call is handled by `method_missing`, it invokes each block, one at a time, and the block runs to completion before the next block starts.

The implementation of `level` is `@broadcasts.map(&:level).min`, so during the first block, the second logger hasn't had `with_level(:error)` called yet, so its log level is still 1. `@boardcasts.map(&:level)` returns `[3,1]`, and `min` takes the `1`. This leads to some output also getting logged even though the level is below the currently set level.